client/me/form-base/index.js

```diff
@@ -90,9 +93,11 @@ export default {
 					this.props.markSaved && this.props.markSaved();
 
 					if ( this.state && this.state.redirect ) {
-						// Sometimes changes in settings require a url refresh to update the UI.
-						// For example when the user changes the language.
-						window.location = this.state.redirect + '?updated=success';
+						user.clear( () => {
+							// Sometimes changes in settings require a url refresh to update the UI.
+							// For example when the user changes the language.
+							window.location = this.state.redirect + '?updated=success';
+						} );
```

Comment:
Shouldn't we update [other](https://github.com/Automattic/wp-calypso/blob/b855b7a8a7e9f6a24e4fc87e8fe251cfd10603fd/client/login/magic-login/handle-emailed-link-form.jsx#L123-L128) [calls](https://github.com/Automattic/wp-calypso/blob/2785e2e0f503ee4657409d8d5509c31586ee41d2/client/blocks/login/index.jsx#L124-L129) to use a callback as well?

Reply:
We can 👍 . Note that `persist-redux` is true in all environments so `localforage.clear( undefined );` was called in all those cases already, so although it does make the code a tiny bit safer it's not required by this PR.